### ISSUE TITLE
Rise of Nations: Extended Edition (287450)

### ISSUE THREAD

**Issue by lej1337** [ValveSoftware/Proton#298]
Currently doesn't launch via Steam Play on Arch Linux. Switching to the oldest version of the beta launches a launcher that is frozen and blank. Couldn't wishlist report because game is already purchased. So doing it here. Any help would be greatly appreciated!

**Comment by kisak-valve** [comment]
Hello @lej1337, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by lej1337** [comment]
https://gist.github.com/lej1337/fe6cac90ebf87809634aed175b3a9ce5

**Comment by kisak-valve** [comment]
You may have a better experience with mesa-git from the AUR.

**Comment by d10sfan** [comment]
For this one, it immediately crashes after loading. System specs are below.

[steam-287450.log](https://github.com/ValveSoftware/Proton/files/2320313/steam-287450.log)

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    CPU Family:  0x6
    CPU Model:  0x5e
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  4200 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Ubuntu 18.04.1 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.15.0-32-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11906000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  steam-runtime-beta-release_2018-06-14

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1070/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 396.54
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x10de
    DeviceID:  0x1b81
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1080
    Desktop Resolution: 2560 x 1080
    Primary Display Size: 31.42" x 13.15" (34.06" diag)
                                            79.8cm x 33.4cm (86.5cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 8192 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC1150

Memory:
    RAM:  15988 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  233201 Mb
    Largest Free Hard Disk Block:  193948 Mb
    VR Headset: None detected

Recent Failure Reports:

**Comment by ryanmusante** [comment]
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  2600 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Manjaro Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.18.4-1-MANJARO
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12001000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  <Runtime disabled>

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 980M/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 396.54
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x13d7
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 13.54" x 7.64" (15.51" diag)
                                            34.4cm x 19.4cm (39.4cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 4096 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Intel Haswell HDMI

Memory:
    RAM:  32057 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  448338 Mb
    Largest Free Hard Disk Block:  151219 Mb
    VR Headset: None detected

Recent Failure Reports:

Will not load , crashes immediately after selecting run.
```
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to RunningInstallScript with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 287450, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
Opted-in Controller Mask: 70
Game update: AppID 287450 "", ProcID 29437, IP 0.0.0.0:0
>>> Adding process 29437 for game ID 287450
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
GameAction [AppID 287450, ActionID 1] : LaunchApp changed task to Completed with ""
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 29440 for game ID 287450
>>> Adding process 29441 for game ID 287450
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 29449 for game ID 287450
>>> Adding process 29450 for game ID 287450
>>> Adding process 29455 for game ID 287450
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 29457 for game ID 287450
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 29460 for game ID 287450
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 29471 for game ID 287450
>>> Adding process 29476 for game ID 287450
ERROR: ld.so: object '/home/ryan/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
pid 29453 != 29450, skipping destruction (fork without exec?)
Game removed: AppID 287450 "", ProcID 29437 
No cached sticky mapping in ActivateActionSet.Generating new string page texture 195: 24x256, total string texture memory is 3.72 MB
Generating new string page texture 196: 16x256, total string texture memory is 3.74 MB
Generating new string page texture 197: 256x256, total string texture memory is 4.00 MB

```

**Comment by ryao** [comment]
Running `env WINEPREFIX=$HOME/.local/share/Steam/SteamApps/compatdata/287450/pfx /home/richard/.local/share/Steam/SteamApps/common/Proton\ 3.7/dist/bin/wine "$HOME/.local/share/Steam/SteamApps/common/Rise of Nations/riseofnations.exe"` will work, minus sound. Doing `env WINEPREFIX=$HOME/.local/share/Steam/SteamApps/compatdata/287450/pfx WINEPATH=$HOME/.local/share/Steam/SteamApps/common/Proton\ 3.7/dist/bin/wine winetricks -q directmusic` and then setting overrides in winecfg for dmime, dmsynth, dmusic, dsound, dswave and l3codecx will make sound work.

<details><summary>

However, running `/tmp/proton_run  "$HOME/.local/share/Steam/SteamApps/common/Rise of Nations/riseofnations.exe"` will crash:

</summary><p>

```
$ /tmp/proton_run  "$HOME/.local/share/Steam/SteamApps/common/Rise of Nations/riseofnations.exe"                           
Setting breakpad minidump AppID = 287450
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198003059557 [API loaded no]
info:  Game: riseofnations.exe
info:  DXVK: v0.64-58-g6a5ad8b
warn:  OpenVR: Failed to initialize OpenVR
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
INTEL-MESA: warning: Haswell Vulkan support is incomplete
info:  Intel(R) Haswell Server:
info:    Driver: 17.3.9
info:    Vulkan: 1.0.57
info:    Memory Heap[0]: 
info:      Size: 1536 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0xf
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_10_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_10_0
info:  Enabled device extensions:
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  Game: riseofnations.exe
info:  DXVK: v0.64-58-g6a5ad8b
warn:  OpenVR: Failed to initialize OpenVR
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
INTEL-MESA: warning: Haswell Vulkan support is incomplete
info:  Intel(R) Haswell Server:
info:    Driver: 17.3.9
info:    Vulkan: 1.0.57
info:    Memory Heap[0]: 
info:      Size: 1536 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0xf
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  Enabled device extensions:
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  9b7e4e00-342c-4106-a19f-4f2704f689f0
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_hw.c:1712:(_snd_pcm_hw_open) Invalid value for card
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1024x768
wine: Unhandled page fault on write access to 0x00000000 at address 0x7e42422d (thread 0027), starting debugger...
Unhandled exception: page fault on write access to 0x00000000 in 32-bit code (0x7e42422d).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:7e42422d ESP:0033eec0 EBP:0033eef8 EFLAGS:00210202(  R- --  I   - - - )
 EAX:00000043 EBX:7e475f60 ECX:06d79693 EDX:0000002a
 ESI:00000000 EDI:7e475f0c
Stack dump:
0x0033eec0:  31303363 0000002a 06d79690 7e493ff4
0x0033eed0:  00000000 000000a8 39373234 00000004
0x0033eee0:  09090a22 09090a7b 616e2209 7e493ff4
0x0033eef0:  06d79690 0033ef90 0033ef58 7e425a1c
0x0033ef00:  00000000 0033ef3c 0909090a 09090a7b
0x0033ef10:  30220909 22090922 6f617972 7e475f0c
Backtrace:
=>0 0x7e42422d encodeBase64W+0x10d(in_buf=í▓∩Ç
▒ >*p╣~┼+IbC█]^
               ╣║∙rZ¢â╦ÉΣ) ┴agâò4┼.^L╪2Ç╢∩┼τ█LöÆò9z1ó²M╥∩▄╥║n÷"⌡└Lφzƒ√╨8CΓ√Lƒ°ñ:ñû╝Åú▄╫hÖ"⌠)╫ä-y1╨¬U┐(ó▀cN»≤4∙", in_len=<is not available>, sep="
", out_buf=0x0(nil), out_len=0x33ef3c) [/home/proton/proton/wine/dlls/crypt32/base64.c:328] in crypt32 (0x0033eef8)
  1 0x7e425a1c CryptBinaryToStringW+0x3bb(pbBinary=<couldn't compute location>, cbBinary=<couldn't compute location>, dwFlags=<couldn't compute location>, pszString=<couldn't compute location>, pcchString=<couldn't compute location>) [/home/proton/proton/wine/dlls/crypt32/base64.c:424] in crypt32 (0x0033ef58)
  2 0x010e7d59 in riseofnations (+0xce7d58) (0x0033efbc)
  3 0x010e8093 in riseofnations (+0xce8092) (0x0033f0b0)
  4 0x00462e96 in riseofnations (+0x62e95) (0x0033f130)
  5 0x010e7f55 in riseofnations (+0xce7f54) (0x0033f15c)
  6 0x00464461 in riseofnations (+0x64460) (0x0033f16c)
  7 0x3b407d53 in steam_api (+0x7d52) (0x0033f1a0)
  8 0x3b407c01 in steam_api (+0x7c00) (0x0033f1d4)
  9 0x3b407eec in steam_api (+0x7eeb) (0x0033f200)
  10 0x3b407925 in steam_api (+0x7924) (0x0033f210)
  11 0x3b40a10a in steam_api (+0xa109) (0x0033f220)
  12 0x3b409c77 in steam_api (+0x9c76) (0x0033f240)
  13 0x0100c9f6 in riseofnations (+0xc0c9f5) (0x0033f260)
  14 0x00c56489 in riseofnations (+0x856488) (0x0033f270)
  15 0x011111cb in riseofnations (+0xd111ca) (0x0033f2f0)
  16 0x00c1c52a in riseofnations (+0x81c529) (0x0033f40c)
  17 0x00c24012 in riseofnations (+0x824011) (0x0033f448)
  18 0x00c5cc2f in riseofnations (+0x85cc2e) (0x0033f49c)
  19 0x00c52fd6 in riseofnations (+0x852fd5) (0x0033f524)
  20 0x00c1e894 in riseofnations (+0x81e893) (0x0033fe48)
  21 0x00c1dc55 in riseofnations (+0x81dc54) (0x0033fe74)
  22 0x00c18bb1 in riseofnations (+0x818bb0) (0x0033fec0)
  23 0x7b4611dc call_process_entry+0xb() in kernel32 (0x0033fed8)
  24 0x7b462a36 start_process+0x105(entry=<couldn't compute location>, peb=<couldn't compute location>) [/home/proton/proton/wine/dlls/kernel32/process.c:1099] in kernel32 (0x0033ffd8)
  25 0x7b4611ea start_process_wrapper+0x9() in kernel32 (0x0033ffec)
0x7e42422d encodeBase64W+0x10d [/home/proton/proton/wine/dlls/crypt32/base64.c:328] in crypt32: movw    %ax,0x0(%esi)
Unable to access file '/home/proton/proton/wine/dlls/crypt32/base64.c'
Modules:
Module  Address                 Debug info      Name (243 modules)
PE        350000-  381000       Deferred        protobuf_lite
PE        400000- 18f4000       Export          riseofnations
PE       1900000- 1a25000       Deferred        boringssl
PE       1a30000- 1b1b000       Deferred        crossplaynetlib
PE       1b20000- 1bd4000       Deferred        cell_v140_release_win32_wide_4_0
PE       23d0000- 250d000       Deferred        steam
PE       2a60000- 2b76000       Deferred        d3dgl
PE       2b80000- 2f1b000       Deferred        d3dcompiler_47
PE       2f20000- 2f99000       Deferred        openvr_api_dxvk
PE       4810000- 48e7000       Deferred        dssl
PE      10000000-104c1000       Deferred        cpprest140_2_9
PE      3b400000-3b439000       Export          steam_api
PE      6a540000-6a73e000       Deferred        d3d11
PE      6f400000-6f596000       Deferred        dxgi
ELF     76276000-7629a000       Deferred        dmsynth<elf>
  \-PE  76280000-7629a000       \               dmsynth
ELF     7629a000-76385000       Deferred        libasound.so.2
ELF     76385000-763b7000       Deferred        winealsa<elf>
  \-PE  76390000-763b7000       \               winealsa
ELF     763b7000-76443000       Deferred        libvorbisenc.so.2
ELF     76ea7000-76ed5000       Deferred        libvorbis.so.0
ELF     76ed5000-76f1a000       Deferred        libflac.so.8
ELF     7771c000-77732000       Deferred        midimap<elf>
  \-PE  77720000-77732000       \               midimap
ELF     77732000-7773a000       Deferred        libogg.so.0
ELF     7773a000-77755000       Deferred        libnsl.so.1
ELF     77755000-7775c000       Deferred        libasyncns.so.0
ELF     7775c000-777ce000       Deferred        libsndfile.so.1
ELF     777ce000-777d9000       Deferred        libwrap.so.0
ELF     777d9000-77823000       Deferred        libdbus-1.so.3
ELF     77823000-7788a000       Deferred        libpulsecommon-1.1.so
ELF     7788a000-77893000       Deferred        libjson.so.0
ELF     77893000-778e1000       Deferred        libpulse.so.0
ELF     778f1000-7790a000       Deferred        msacm32<elf>
  \-PE  77900000-7790a000       \               msacm32
ELF     7790a000-7792d000       Deferred        mmdevapi<elf>
  \-PE  77910000-7792d000       \               mmdevapi
ELF     7792d000-77958000       Deferred        dmusic<elf>
  \-PE  77930000-77958000       \               dmusic
ELF     77958000-779a3000       Deferred        dsound<elf>
  \-PE  77960000-779a3000       \               dsound
ELF     779a3000-779e1000       Deferred        dmime<elf>
  \-PE  779b0000-779e1000       \               dmime
ELF     779e1000-77a0e000       Deferred        dmloader<elf>
  \-PE  779f0000-77a0e000       \               dmloader
ELF     77a0e000-77d12000       Deferred        libvulkan_intel.so
ELF     77f12000-78320000       Deferred        vrclient.so
ELF     7849a000-784d1000       Deferred        msctf<elf>
  \-PE  784a0000-784d1000       \               msctf
ELF     784d1000-784df000       Deferred        libwayland-client.so.0
ELF     784df000-79e89000       Deferred        libicudata.so.60
ELF     79e89000-7b400000       Deferred        steamclient.so
ELF     7b400000-7b7ea000       Dwarf           kernel32<elf>
  \-PE  7b410000-7b7ea000       \               kernel32
ELF     7b7ec000-7b7f5000       Deferred        libffi.so.6
ELF     7b7f5000-7b834000       Deferred        rsaenh<elf>
  \-PE  7b800000-7b834000       \               rsaenh
ELF     7b834000-7b849000       Deferred        libdrm.so.2
ELF     7b849000-7b866000       Deferred        libxcb-glx.so.0
ELF     7b866000-7b884000       Deferred        libglapi.so.0
ELF     7b884000-7b8f6000       Deferred        libgl.so.1
ELF     7b8f6000-7b90a000       Deferred        api-ms-win-core-file-l2-1-1<elf>
  \-PE  7b900000-7b90a000       \               api-ms-win-core-file-l2-1-1
ELF     7b90a000-7b91e000       Deferred        api-ms-win-core-localization-l1-2-1<elf>
  \-PE  7b910000-7b91e000       \               api-ms-win-core-localization-l1-2-1
ELF     7b91e000-7b948000       Deferred        libpng12.so.0
ELF     7b948000-7ba00000       Deferred        msvcrt<elf>
  \-PE  7b960000-7ba00000       \               msvcrt
ELF     7bc00000-7bd0f000       Deferred        ntdll<elf>
  \-PE  7bc10000-7bd0f000       \               ntdll
ELF     7bd13000-7bd19000       Deferred        libxcb-dri2.so.0
ELF     7bd19000-7bd1d000       Deferred        libxdamage.so.1
ELF     7bd1d000-7bd31000       Deferred        api-ms-win-core-fibers-l1-1-1<elf>
  \-PE  7bd20000-7bd31000       \               api-ms-win-core-fibers-l1-1-1
ELF     7bd31000-7be00000       Deferred        libkrb5.so.3
ELF     7c000000-7c004000       Deferred        <wine-loader>
ELF     7c007000-7c00f000       Deferred        libxcb-sync.so.1
ELF     7c00f000-7c023000       Deferred        api-ms-win-core-synch-l1-2-0<elf>
  \-PE  7c010000-7c023000       \               api-ms-win-core-synch-l1-2-0
ELF     7c023000-7c070000       Deferred        libvulkan.so.1
ELF     7c070000-7c0af000       Deferred        winevulkan<elf>
  \-PE  7c080000-7c0af000       \               winevulkan
ELF     7c0af000-7c0c7000       Deferred        vulkan-1<elf>
  \-PE  7c0b0000-7c0c7000       \               vulkan-1
ELF     7c0c7000-7c0fc000       Deferred        secur32<elf>
  \-PE  7c0d0000-7c0fc000       \               secur32
ELF     7c405000-7c436000       Deferred        netapi32<elf>
  \-PE  7c410000-7c436000       \               netapi32
ELF     7c438000-7c43c000       Deferred        libxcb-present.so.0
ELF     7c43c000-7c451000       Deferred        schannel<elf>
  \-PE  7c440000-7c451000       \               schannel
ELF     7c451000-7c48f000       Deferred        libgssapi_krb5.so.2
ELF     7c48f000-7c4b7000       Deferred        libk5crypto.so.3
ELF     7c4b7000-7c4fa000       Deferred        libxslt.so.1
ELF     7c4fa000-7c596000       Deferred        urlmon<elf>
  \-PE  7c510000-7c596000       \               urlmon
ELF     7c596000-7c751000       Deferred        libicuuc.so.60
ELF     7c751000-7c8d0000       Deferred        libxml2.so.2
ELF     7c8d0000-7c9b4000       Deferred        msxml3<elf>
  \-PE  7c8e0000-7c9b4000       \               msxml3
ELF     7c9b4000-7cb00000       Deferred        libx11.so.6
ELF     7cc01000-7cc0a000       Deferred        libxcb-xfixes.so.0
ELF     7cc0a000-7cc10000       Deferred        libxcb-dri3.so.0
ELF     7cc10000-7cc2a000       Deferred        kerberos<elf>
  \-PE  7cc20000-7cc2a000       \               kerberos
ELF     7cc2a000-7cc56000       Deferred        liblzma.so.5
ELF     7cd52000-7cd55000       Deferred        libx11-xcb.so.1
ELF     7cd55000-7cd59000       Deferred        libkeyutils.so.1
ELF     7cd59000-7cd62000       Deferred        libkrb5support.so.0
ELF     7cd62000-7cd67000       Deferred        libcom_err.so.2
ELF     7cdad000-7cdb4000       Deferred        libxfixes.so.3
ELF     7cdb4000-7cdc0000       Deferred        libxcursor.so.1
ELF     7cdc0000-7cdd3000       Deferred        libxi.so.6
ELF     7cdd3000-7ce00000       Deferred        libxcb.so.1
ELF     7d100000-7d103000       Deferred        libxshmfence.so.1
ELF     7d105000-7d111000       Deferred        libxrender.so.1
ELF     7d111000-7d115000       Deferred        libxcomposite.so.1
ELF     7d115000-7d11c000       Deferred        libxxf86vm.so.1
ELF     7d11c000-7d120000       Deferred        libxinerama.so.1
ELF     7d120000-7d127000       Deferred        libxdmcp.so.6
ELF     7d127000-7d12c000       Deferred        libxau.so.6
ELF     7d12c000-7d142000       Deferred        libxext.so.6
ELF     7d142000-7d1d4000       Deferred        winex11<elf>
  \-PE  7d150000-7d1d4000       \               winex11
ELF     7d1d4000-7d1f0000       Deferred        libgcc_s.so.1
ELF     7d1f0000-7d1f7000       Deferred        libnss_dns.so.2
ELF     7d1f7000-7d20b000       Deferred        psapi<elf>
  \-PE  7d200000-7d20b000       \               psapi
ELF     7d20b000-7d41b000       Deferred        lsteamclient<elf>
  \-PE  7d2c0000-7d41b000       \               lsteamclient
ELF     7d41b000-7d443000       Deferred        mfplat<elf>
  \-PE  7d420000-7d443000       \               mfplat
ELF     7d443000-7d467000       Deferred        libgpg-error.so.0
ELF     7d467000-7d479000       Deferred        libp11-kit.so.0
ELF     7d479000-7d4ff000       Deferred        libgcrypt.so.11
ELF     7d4ff000-7d511000       Deferred        libtasn1.so.3
ELF     7d511000-7d5d9000       Deferred        libgnutls.so.26
ELF     7d5d9000-7d5fd000       Deferred        imm32<elf>
  \-PE  7d5e0000-7d5fd000       \               imm32
ELF     7d68e000-7d6c0000       Deferred        libexpat.so.1
ELF     7d6c0000-7d70b000       Deferred        libfontconfig.so.1
ELF     7d70b000-7d74a000       Deferred        libpng16.so.16
ELF     7d74a000-7d75c000       Deferred        libbz2.so.1
ELF     7d75c000-7d82c000       Deferred        libfreetype.so.6
ELF     7d82c000-7d856000       Deferred        iphlpapi<elf>
  \-PE  7d830000-7d856000       \               iphlpapi
ELF     7d856000-7d86a000       Deferred        shfolder<elf>
  \-PE  7d860000-7d86a000       \               shfolder
ELF     7d86a000-7d892000       Deferred        mpr<elf>
  \-PE  7d870000-7d892000       \               mpr
ELF     7d892000-7d90b000       Deferred        wininet<elf>
  \-PE  7d8a0000-7d90b000       \               wininet
ELF     7d90b000-7d980000       Deferred        shlwapi<elf>
  \-PE  7d920000-7d980000       \               shlwapi
ELF     7d980000-7dc20000       Deferred        shell32<elf>
  \-PE  7d990000-7dc20000       \               shell32
ELF     7dc20000-7dc3e000       Deferred        msdmo<elf>
  \-PE  7dc30000-7dc3e000       \               msdmo
ELF     7dc3e000-7dc56000       Deferred        userenv<elf>
  \-PE  7dc40000-7dc56000       \               userenv
ELF     7dc56000-7de50000       Deferred        libcrypto.so.1.0.0
ELF     7de50000-7dec0000       Deferred        libssl.so.1.0.0
ELF     7dec0000-7ded8000       Deferred        libresolv.so.2
ELF     7ded8000-7dee9000       Deferred        liblber-2.4.so.2
ELF     7dee9000-7df43000       Deferred        libldap_r-2.4.so.2
ELF     7df43000-7df9d000       Deferred        wldap32<elf>
  \-PE  7df50000-7df9d000       \               wldap32
ELF     7df9d000-7dfb6000       Deferred        api-ms-win-crt-math-l1-1-0<elf>
  \-PE  7dfa0000-7dfb6000       \               api-ms-win-crt-math-l1-1-0
ELF     7dfb6000-7dfca000       Deferred        api-ms-win-crt-filesystem-l1-1-0<elf>
  \-PE  7dfc0000-7dfca000       \               api-ms-win-crt-filesystem-l1-1-0
ELF     7dfca000-7dfde000       Deferred        api-ms-win-crt-conio-l1-1-0<elf>
  \-PE  7dfd0000-7dfde000       \               api-ms-win-crt-conio-l1-1-0
ELF     7dfde000-7dff2000       Deferred        api-ms-win-crt-utility-l1-1-0<elf>
  \-PE  7dfe0000-7dff2000       \               api-ms-win-crt-utility-l1-1-0
ELF     7dff2000-7e006000       Deferred        api-ms-win-crt-environment-l1-1-0<elf>
  \-PE  7e000000-7e006000       \               api-ms-win-crt-environment-l1-1-0
ELF     7e006000-7e01a000       Deferred        api-ms-win-crt-time-l1-1-0<elf>
  \-PE  7e010000-7e01a000       \               api-ms-win-crt-time-l1-1-0
ELF     7e01a000-7e02e000       Deferred        api-ms-win-crt-heap-l1-1-0<elf>
  \-PE  7e020000-7e02e000       \               api-ms-win-crt-heap-l1-1-0
ELF     7e02e000-7e042000       Deferred        api-ms-win-crt-locale-l1-1-0<elf>
  \-PE  7e030000-7e042000       \               api-ms-win-crt-locale-l1-1-0
ELF     7e042000-7e058000       Deferred        api-ms-win-crt-convert-l1-1-0<elf>
  \-PE  7e050000-7e058000       \               api-ms-win-crt-convert-l1-1-0
ELF     7e058000-7e06e000       Deferred        api-ms-win-crt-stdio-l1-1-0<elf>
  \-PE  7e060000-7e06e000       \               api-ms-win-crt-stdio-l1-1-0
ELF     7e06e000-7e084000       Deferred        api-ms-win-crt-string-l1-1-0<elf>
  \-PE  7e070000-7e084000       \               api-ms-win-crt-string-l1-1-0
ELF     7e084000-7e09a000       Deferred        api-ms-win-crt-runtime-l1-1-0<elf>
  \-PE  7e090000-7e09a000       \               api-ms-win-crt-runtime-l1-1-0
ELF     7e09a000-7e0b0000       Deferred        vcruntime140<elf>
  \-PE  7e0a0000-7e0b0000       \               vcruntime140
ELF     7e0b0000-7e0e9000       Deferred        ws2_32<elf>
  \-PE  7e0c0000-7e0e9000       \               ws2_32
ELF     7e0e9000-7e108000       Deferred        concrt140<elf>
  \-PE  7e0f0000-7e108000       \               concrt140
ELF     7e108000-7e1f3000       Deferred        msvcr120<elf>
  \-PE  7e120000-7e1f3000       \               msvcr120
ELF     7e1f3000-7e2ed000       Deferred        ucrtbase<elf>
  \-PE  7e210000-7e2ed000       \               ucrtbase
ELF     7e2ed000-7e40c000       Deferred        msvcp140<elf>
  \-PE  7e330000-7e40c000       \               msvcp140
ELF     7e40c000-7e4da000       Dwarf           crypt32<elf>
  \-PE  7e420000-7e4da000       \               crypt32
ELF     7e4da000-7e4fb000       Deferred        bcrypt<elf>
  \-PE  7e4e0000-7e4fb000       \               bcrypt
ELF     7e4fb000-7e511000       Deferred        httpapi<elf>
  \-PE  7e500000-7e511000       \               httpapi
ELF     7e511000-7e641000       Deferred        oleaut32<elf>
  \-PE  7e530000-7e641000       \               oleaut32
ELF     7e641000-7e65f000       Deferred        jsproxy<elf>
  \-PE  7e650000-7e65f000       \               jsproxy
ELF     7e65f000-7e69b000       Deferred        winhttp<elf>
  \-PE  7e670000-7e69b000       \               winhttp
ELF     7e69b000-7e6b4000       Deferred        libz.so.1
ELF     7e6b4000-7e71a000       Deferred        dbghelp<elf>
  \-PE  7e6c0000-7e71a000       \               dbghelp
ELF     7e71a000-7e745000       Deferred        msacm32<elf>
  \-PE  7e720000-7e745000       \               msacm32
ELF     7e745000-7e7c5000       Deferred        rpcrt4<elf>
  \-PE  7e750000-7e7c5000       \               rpcrt4
ELF     7e7c5000-7e91c000       Deferred        ole32<elf>
  \-PE  7e7e0000-7e91c000       \               ole32
ELF     7e91c000-7e936000       Deferred        version<elf>
  \-PE  7e920000-7e936000       \               version
ELF     7e936000-7e9ae000       Deferred        advapi32<elf>
  \-PE  7e940000-7e9ae000       \               advapi32
ELF     7e9ae000-7eadb000       Deferred        gdi32<elf>
  \-PE  7e9c0000-7eadb000       \               gdi32
ELF     7eadb000-7ecd9000       Deferred        user32<elf>
  \-PE  7eaf0000-7ecd9000       \               user32
ELF     7ecd9000-7ed91000       Deferred        winmm<elf>
  \-PE  7ece0000-7ed91000       \               winmm
ELF     7ed91000-7ed9f000       Deferred        libnss_files.so.2
ELF     7ef3b000-7f000000       Deferred        libm.so.6
ELF     f7b80000-f7b8a000       Deferred        librt.so.1
ELF     f7b8c000-f7b91000       Deferred        libdl.so.2
ELF     f7b91000-f7d6f000       Deferred        libc.so.6
ELF     f7d6f000-f7d8f000       Deferred        libpthread.so.0
ELF     f7d8f000-f7f46000       Dwarf           libwine.so.1
ELF     f7f48000-f7f6f000       Deferred        ld-linux.so.2
ELF     f7f72000-f7f73000       Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
0000000c services.exe
        00000020    0
        0000001b    0
        00000011    0
        0000000e    0
        0000000d    0
0000000f winedevice.exe
        00000018    0
        00000015    0
        00000014    0
        00000010    0
00000019 plugplay.exe
        0000001d    0
        0000001c    0
        0000001a    0
0000001e winedevice.exe
        00000025    0
        00000022    0
        00000021    0
        0000001f    0
00000026 (D) Z:\home\richard\.local\share\Steam\SteamApps\common\Rise of Nations\riseofnations.exe
        0000003a   15
        00000039   15
        00000038    0
        00000037    0
        00000036    0
        00000033    0
        0000002a    0
        00000029    0
        00000028    0
        00000027    0 <==
0000002c explorer.exe
        00000032    0
        00000031    0
        00000030    0
        0000002d    0
System information:
    Wine build: wine-3.7
    Platform: i386 (WOW64)
    Version: Windows 7
    Host system: Linux
    Host version: 4.18.0-rc8
```

</p></details>

<details><summary>Here is my system information:</summary><p>

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Xeon(R) CPU E3-1276 v3 @ 3.60GHz
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  4000 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "NAME=Gentoo" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.18.0-rc8
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11905000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime-beta-release_2018-06-14

Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Haswell Server x86/MMX/SSE2
    Driver Version:  3.0 Mesa 17.3.9
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID Not Detected
    DeviceID Not Detected
    Revision Not Detected
    Number of Monitors:  1
    Number of Video Cards Not Detected
    Primary Display Resolution:  1920 x 1200
    Desktop Resolution: 1920 x 1200
    Primary Display Size: 20.43" x 12.76" (24.06" diag)
                                            51.9cm x 32.4cm (61.1cm diag)
    Primary VRAM Not Detected

Sound card:
    Audio device: USB Mixer

Memory:
    RAM:  32119 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  254048 Mb
    Largest Free Hard Disk Block:  94045 Mb
    VR Headset: None detected

Recent Failure Reports:


```

</p></details>

Doing `env WINEPREFIX=$HOME/.local/share/Steam/SteamApps/compatdata/287450/pfx WINEPATH=$HOME/.local/share/Steam/SteamApps/common/Proton\ 3.7/dist/bin/wine winetricks -q crypt32` and then setting an override in winecfg for crypt32 will make `/tmp/proton_run  "$HOME/.local/share/Steam/SteamApps/common/Rise of Nations/riseofnations.exe"` work. You will get sound if you used winetricks to install directmusic and do the aforementioned overrides too.

<details><summary>Here is what happens when doing that after the overrides:</summary><p>

```
$ /tmp/proton_run  "$HOME/.local/share/Steam/SteamApps/common/Rise of Nations/riseofnations.exe"
wine: cannot find L"C:\\windows\\system32\\winemenubuilder.exe"
Setting breakpad minidump AppID = 287450
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198003059557 [API loaded no]
info:  Game: riseofnations.exe
info:  DXVK: v0.64-58-g6a5ad8b
warn:  OpenVR: Failed to initialize OpenVR
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
INTEL-MESA: warning: Haswell Vulkan support is incomplete
info:  Intel(R) Haswell Server:
info:    Driver: 17.3.9
info:    Vulkan: 1.0.57
info:    Memory Heap[0]: 
info:      Size: 1536 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0xf
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_10_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_10_0
info:  Enabled device extensions:
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  Game: riseofnations.exe
info:  DXVK: v0.64-58-g6a5ad8b
warn:  OpenVR: Failed to initialize OpenVR
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
INTEL-MESA: warning: Haswell Vulkan support is incomplete
info:  Intel(R) Haswell Server:
info:    Driver: 17.3.9
info:    Vulkan: 1.0.57
info:    Memory Heap[0]: 
info:      Size: 1536 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0xf
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  Enabled device extensions:
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  9b7e4e00-342c-4106-a19f-4f2704f689f0
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_hw.c:1712:(_snd_pcm_hw_open) Invalid value for card
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_speexrate_best.so
ALSA lib /var/tmp/portage/media-libs/alsa-lib-1.1.2/work/alsa-lib-1.1.2/src/pcm/pcm_rate.c:1324:(snd_pcm_rate_open) Cannot find rate converter
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1024x768
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1200
```

</p></details>

If we someone could fix the bug in crypt32 and then work on sound support, this should work fairly well. Unfortunately, multiplayer is broken.

Credit for making the music work goes to this guide:

https://drive.google.com/file/d/0B1lsK8727OpLODI0OTQzMzAtYzBjMC00OTIzLWE1NzAtOGIwYTBlMWEzMTA3/view

Although I am the author, so crediting myself might be a little self serving. However, I honestly would not have had any clue had it not been for that old guide. I made it as part of an assignment for a Computer Science writing class when I was in college. It had been the result after much trial and error with trying various tips that I found on winehq and elsewhere.

That said, I guess the rename trick where you replace patriots.exe with riseofnations.exe to bypass the launcher like people are doing in #72 would allow Steam itself to launch this once directmusic and crypt32 with the winecfg overrides I mentioned are in place. I.e.:

```
$ mv $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe{,.bak}
$ ln -s $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/riseofnations.exe $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe
```

**Comment by ryao** [comment]
Here is the corresponding winehq issue for the segfault in crypt32:

https://bugs.winehq.org/show_bug.cgi?id=44405

**Comment by ryao** [comment]
I opened a pull request with a fix at ValveSoftware/wine#15. Once that is merged, all that people will need to do is to run these two commands:

```
mv $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe{,.bak}
ln -s $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/riseofnations.exe $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe
```

Then it will work, minus sound. Fixing sound requires installing directmusic via winetricks and then setting overrides in winecfg for dmime, dmsynth, dmusic, dsound, dswave and l3codecx

**Comment by cromat** [comment]
> I opened a pull request with a fix at [ValveSoftware/wine#15](https://github.com/ValveSoftware/wine/pull/15). Once that is merged, all that people will need to do is to run these two commands:
> 
> ```
> mv $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe{,.bak}
> ln -s $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/riseofnations.exe $HOME/.local/share/Steam/SteamApps/common/Rise\ of\ Nations/patriots.exe
> ```
> Then it will work, minus sound. Fixing sound requires installing directmusic via winetricks and then setting overrides in winecfg for dmime, dmsynth, dmusic, dsound, dswave and l3codecx

I cannot get sound working. When I install and override all libraries and start the game I hear some short noises but when the menu appears there is no sound. Also I have installed l3codecx but I don't get it listed in libraries dropdown list.

**Comment by ryao** [comment]
@cromat Things seem to have changed. I can no longer get sound working with a fresh prefix. Also, Valve seems to have changed the Steam directory location. Here are some fresh instructions:

```
# Wipe out wine prefix directory to get a fresh one.
rm -rf /home/richard/.steam/steam/steamapps/compatdata/287450

# Set launch options: PROTON_DUMP_DEBUG_COMMANDS=1 %command%
# Run Rise of Nations, which will then fail.
steam -applaunch 287450

# You may remove the launch options after doing that.

# Disable launcher
mv ${HOME}/.steam/steam/steamapps/common/Rise\ of\ Nations/patriots.exe{,.bak}
ln -s riseofnations.exe ${HOME}/.steam/steam/steamapps/common/Rise\ of\ Nations/patriots.exe

# Fetch winetricks and protontricks
wget https://raw.githubusercontent.com/Sirmentio/protontricks/master/protontricks
wget https://raw.githubusercontent.com/Winetricks/winetricks/master/src/winetricks

# Set them as executable
chmod u+x "${PWD}/protontricks" "${PWD}/winetricks"

# Add crypt32 workaround
WINESERVER="${HOME}/.local/share/Steam/SteamApps/common/Proton 3.7/dist/bin/wineserver" WINETRICKS="${PWD}/winetricks" ${PWD}/protontricks 287450 crypt32

# Enjoy
steam -applaunch 287450
```

Trying to make sound work on the latest proton seems to break the game. :/

**Comment by ryanmusante** [comment]
@ryao So, you're loading riseofnations.exe rather than patriots.exe default? Is the expansion loading?

**Comment by ryao** [comment]
@ryanmusante I am not interested in the expansion.

**Comment by ryanmusante** [comment]
@ryao it introduces too much that causes imbalance?

**Comment by ryao** [comment]
It just never appealed to me. I liked the original and I did not want more content.

**Comment by ryao** [comment]
I figured it out. The hang is caused by an incompatibility between Wine, directsound and PulseAudio. Setting `sound=alsa` via winetricks makes it work.

It worked for me (and no one else) because I initially set it up on my system with Intel graphics that lacks pulseaudio. It did not work for me when I tried again on my system with Nvidia graphics because it has pulseaudio. I had a thought that I could figure out what was going wrong by comparing the two. The differences were ALSA and Intel graphics vs PulseAudio and Nvidia graphics. Glog78 of the VKx discord suggested trying `sound=alsa` after mentioning it to him, which I found made it work. :)

This also nicely explains why it worked for me on the older Proton, but not the newer Proton, despite Proton's audio code had not been patched at all between the versions.

**Comment by d10sfan** [comment]
When I launch the game (after the crypt32 workaround), the main menu will flicker. Is there a way to fix that?

**Comment by ghost** [comment]
> When I launch the game (after the crypt32 workaround), the main menu will flicker. Is there a way to fix that?

Add `PROTON_USE_WINED3D11=1 %command%` to start options.

**Comment by d10sfan** [comment]
@nohatssir Thanks that fixed the flickering. 

Any way to get sound working? I tried the directmusic route, but that caused the game to stop working.

**Comment by ryao** [comment]
SteamVR has a fix for the crypt32 workaround.

https://github.com/ValveSoftware/wine/issues/26

@d10sfan Use protontricks to set sound=alsa and directmusic.

**Comment by d10sfan** [comment]
@ryao Worked, thanks!

**Comment by ryao** [comment]
This game has an issue with the Intro Movies not playing. It is easy to mistake as just the game being slow because it displays a black screen for a while before finally starting. Setting `SkipIntroMovies=1` in `~/.local/share/Steam/steamapps/compatdata/287450/pfx/drive_c/users/steamuser/Application Data/Microsoft Games/Rise of Nations/rise2.ini` fixes this and makes the game start instantly.

**Comment by ryao** [comment]
This game performs better with `PROTON_USE_WINED3D11=1` on Intel hardware. Also, there is graphical corruption on both Intel and Nvidia hardware in the text showing resources as shown here:

![image](https://user-images.githubusercontent.com/1386373/47822646-3661fb00-dd3b-11e8-9ba3-2c684b7ab247.png)

Interesting, the one on the left only has corruption when it flashes yellow or red. The corruption on the right text is less common for unit count and wonder count doesn't always happen.

**Comment by ryao** [comment]
In case it helps the wined3d developers, here is a save file that can be used to reproduce the problem:
[savefile.tar.gz](https://github.com/ValveSoftware/Proton/files/2536613/savefile.tar.gz)

**Comment by kisak-valve** [comment]
Rise of Nations: Extended Edition Sound Problem

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1947.
@hsnnsnc posted on 2018-11-11T00:07:55:

# Compatibility Report
- Name of the game with compatibility issues: Rise of Nations: Extended Edition
- Steam AppID of the game: 287450

## System Information
- GPU: GeForce GTX 1050 Ti
- Driver/LLVM version: nvidia 410.73
- Kernel version: 4.18.0-10-generic
- Proton version: 3.16-4 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

 ISSUE
Rise of Nations: Extended Edition of has no sound.

OS
Ubuntu 18.10

STEPS

Select Library from the main menu
Select All Games
Select Rise of Nations: Extended Edition
Click Play

RESULTS
Rise of Nations: Extended Edition has been opened but there is no sound.

EXPECTED
Rise of Nations: Extended Edition should be the sounds of the game.

**Comment by lawremi** [comment]
After setting `PROTON_USE_WINED3D11=1` and installing directmusic with `sound=alsa`, the game starts, with sound, but it always freezes at the "Setup Animals" step when launching a scenario. There are no clues as to why in the console output.  This is with Proton 3.16-4 Beta.

**Comment by nottux** [comment]
> After setting `PROTON_USE_WINED3D11=1` and installing directmusic with `sound=alsa`, the game starts, with sound, but it always freezes at the "Setup Animals" step when launching a scenario. There are no clues as to why in the console output. This is with Proton 3.16-4 Beta.

same thing as yours, can't start a new game, game just freezes at loading screen
i can play quick match, but tutorial and scenario is not launching with hogging up one cpu at %100

**Comment by lawremi** [comment]
I ran `gdb_attach` before loading a tutorial and it looks like there is an issue with the scenario compiler. For example, attempting to load the first tutorial yields:

```
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 345): Invalid call, params specified do not match any of 1 available \"give_good\" function(s) or script(s)")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 345): Invalid call, params specified do not match any of 1 available \"give_good\" function(s) or script(s)")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 345): Syntax Error: \tgive_good(Player, \"Metal\", 100);\r")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 345): Syntax Error: \tgive_good(Player, \"Metal\", 100);\r")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 348): Syntax Error: \tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 348): Syntax Error: \tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 349): Syntax Error: \tif (chance)")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 349): Syntax Error: \tif (chance)")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 350): Syntax Error: \t\tif (chance)")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 350): Syntax Error: \t\tif (chance)")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 351): Syntax Error: \t\t\tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 351): Syntax Error: \t\t\tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("\n")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 354): Syntax Error: \t\t\tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("ERROR: tut_lib.bhs (Line 354): Syntax Error: \t\t\tcreate_unit(Player, x,")
00000029:0000002a: output debug string ("\n")
```

But the campaign does seem to work.

**Comment by nottux** [comment]
version  1.10 works with camping and scenario

**Comment by lawremi** [comment]
The tutorial works in 1.10, as well. The `tut_lib.bhs` file is exactly the same between 1.10 and 1.20, except 1.20 lacks a `\r\n` at the end of the file. The code for the function containing the syntax error is identical (even in raw bytes) between the two. Perhaps someone should confirm this not an issue on Windows, as well.

**Comment by hsnnsnc** [comment]
Sound, scenario and camping are still not working.

**Comment by Ahmed-E-86** [comment]
**A possible fix for the game:**
https://github.com/doitsujin/dxvk/issues/728#issuecomment-459839962

**Comment by craftyguy** [comment]
Requiring random DLLs from the internet (lol) or a Windows install is hardly a 'fix', it's more of a 'workaround' or 'hack'

> You can search for the file with: mfplat.dll version 12.0.7601.23471 windows 7 64 bit or copy the file from your Windows 7 SP1 operating system

**Comment by ryao** [comment]
@ahmed-elsayed2017 That would just make the introduction movies work. If you wait long enough, they finish and you can play the game. Alternatively, you can modify the game’s ini file to disable those movies and the game starts almost instantaneously.

**Comment by pisandelli** [comment]
I had no success trying to run it. Just installed last version of Steam (for Linux), Proton, protontricks... the game's menu shows but when I hit play an error about DirectX10 is thrown (That I must have DX10 or +).
Setting `PROTON_USE_WINED3D11=1` actually `PROTON_USE_WINED3D : 1` didn't worked either.
After a day searching and trying without any success... I just uninstalled everything and will keep waiting until a new breath of hope comes.

- UBUNTU 19.04  
- Intel i7-4510U  
- AMD Radeon r7 m260  
- Steam, Proton, Protontricks... all at their latest version....

**Comment by kisak-valve** [comment]
Hello @pisandelli, for your system, giving https://github.com/ValveSoftware/Proton/wiki/For-AMD-users-having-issues-with-non-OpenGL-games a read and making sure you have vulkan drivers installed for your video card with something like `apt policy mesa-vulkan-drivers mesa-vulkan-drivers:i386` should get the game closer to working.

Since the suggestions to set `PROTON_USE_WINED3D11=1`, DXVK has grown DX10 support so you should not need it, but if you want to use the DirectX to OpenGL render path, it is most easily done by adding `PROTON_USE_WINED3D=1 %command%` to the game's launch options in Steam.

**Comment by pisandelli** [comment]
Thank you @kisak-valve I'll give another try soon and post the result here!
Cheers

**Comment by pisandelli** [comment]
Thank you @kisak-valve I'll give another try soon and post the result here!
Cheers

**Comment by pisandelli** [comment]
Oh man! @kisak-valve your tip was right in the bull eye! Thanks! Just installed Vulkan drivers, and follow the steps mentioned in the link. First without success, but then I just installed mesa-vulkan-drivers:i386 and could play the game. First, the tutorial freezed the game and CPU usage was 100%... but then I just made my old configs and now play my solo Game.
I had to run `protontricks 287450 dsound directmusic` to get sound working.
Thanks a lot!

**Comment by pisandelli** [comment]
@kisak-valve one problem... the gpu overheated... so badly that my notebook turned off...
In game's options I disabled most of features as vsync, antialias, and some other render options. I was able to play for almost 1,5 hour, but my notebook still heats above normal. I had to use a Fan right on the notebook to help cooling. I guess this situation might be very harmful to my hardware.
Please, let me know if there's any kind of configuration in Proton that could help.
I'll try to find some amdgpu config too.
Thanks

**Comment by NerosTie** [comment]
My sound is still crackling with `protontricks 287450 dsound directmusic` an proton 4.2-5. Any idea why it works for the others but not for me?

**Comment by data4pass** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Rise of Nations: Extended Edition
- Steam AppID of the game: 287450

## System Information
- GPU: Intel(R) Pentium(R) CPU 2020M @ 2.40GHz
- Driver/LLVM version: Intel Open Source Technology Center Mesa DRI Intel(R) Ivybridge Mobile x86/MMX/SSE2
- Kernel version: 5.0.0-15-generic
- Proton version: 4.2-9, [this is the protontricks version that I use](https://github.com/Matoking/protontricks)
- OS:  Pop!_OS 19.04 (64 bit)
- Full system information: [Gist](https://gist.github.com/data4pass/57d43129bb0f34918140e9698efbc082)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.

- [x] that I have checked whether there are updates for my system available.



## Symptoms 
I installed Rise of Nations: Extended Edition v1.10 and then installed `directmusic` and set `sound=alsa` using the protontricks GUI. I didn't set any overrides or file renaming but the game launched successfully on my computer with sound working; however I got this graphical bug:

https://i.imgur.com/gZmPJxZ.png

Notice the bug on the buildings' sprites. What might've caused the bug? Is there a way to fix it?

## Reproduction
1. Install the game with Proton 4.2-9.
2. Change the game version to v1.10 (Properties > Betas).
3. Install `directmusic` and set `sound=alsa` using the protontricks GUI.
4. Launch the game.

**Comment by mozo78** [comment]
This is a problem with Proton. With vanilla Wine buildings are looking fine.

**Comment by ryao** [comment]
@data4pass Is there any reason why you are not using proton 4.11-2? That looks like an issue with DXVK. Proton 4.11-2 has a newer version. See if you can reproduce it there.

**Comment by ozyman42** [comment]
Sound still not working with 4.11-12

**Comment by ozyman42** [comment]
Got the sound working w/ the following commands 

```
protontricks 287450 sound=alsa
protontricks 287450 directmusic
protontricks 287450 directplay
```

See https://www.protondb.com/app/287450

**Comment by kaktuspalme** [comment]
On Proton 5.0 sound worked for me correctly without setting sound=alsa. Installation of directmusic was still needed to get audio working.

**Comment by davidak** [comment]
My only issue with Proton 5.0-9 is that i have no audio.

I was able to join the multiplayer lobby, start a game and also got the achivement. So, is `directplay` still needed? Will i get other issues later in the game?

![Screenshot from 2020-07-02 01-29-51](https://user-images.githubusercontent.com/91113/86300600-a8bb6a80-bc03-11ea-8851-260c17d37a36.png)

I tried to install `directmusic`, but it fails. See log: https://gist.github.com/davidak/4493e767cd28752921924f3d2a606245

But audio works now, so i think it is fine. Did you had that too? Is that normal?

**Comment by s-i-e-v-e** [comment]
@kisak-valve

> Rise of Nations: Extended Edition Sound Problem
> 
> Issue transferred from #1947. @hsnnsnc posted on 2018-11-11T00:07:55:
> 

This issue has not been fixed as of Proton 6.3. Still have to resort to the protontricks hack, and even that, sometimes, has problems due to its inability to overwrite symlinks with actual files. Had to implement [another workaround](https://github.com/s-i-e-v-e/docs/blob/master/notes/games/ron_protontricks) on top of it to get sound.

**Comment by pisandelli** [comment]
Hey guys! I was able to run the game smoothly with sound using Proton 5.0 (I was trying with 6.x... without success)
Then just went to the terminal and added the commands  
```
protontricks 287450 sound=alsa
protontricks 287450 directmusic
protontricks 287450 directplay
```

**Comment by lIWAPIl** [comment]
When launching trough steam it starts the launcher. I click play and it gives me a "Could not initialize DirectX! ..." error. I've tried older versions of proton, with no luck. The game launches with wine and plays without sound or multiplayer connectivity trough "riseofnations.exe" located in the installation folder.

OS: Arch LinuxKERNEL: 5.18.3-arch1-1
CPU: AMD Ryzen 5 1600X Six-Core
GPU: NVIDIA GeForce GTX 1050 Ti
GPU DRIVER: NVIDIA 515.48.07
RAM: 16 GB

**Comment by zskamljic** [comment]
Using either wine or latest proton experimental the game stops responding when loading a game (for example, learn to play -> first campaign) the loading process stops about halfway. The music continues to play, but the game never loads.

Using protontricks 287450 directmusic

OS: ArchLinux 6.0.9-arch1-1
CPU: AMD Ryzen 7 3700X 8-Core
GPU: NVIDIA GeForce RTX 2070Ti
GPU Driver: NVIDIA 520.56.06-11
RAM: 32 GB

Looks like the issue is the same as mentioned in [this comment](https://github.com/ValveSoftware/Proton/issues/298#issuecomment-439671589), the game starts otherwise
